test/Plutip.purs

```diff
@@ -336,6 +340,98 @@ suite = do
           logInfo' "Try to spend locked values"
           AlwaysSucceeds.spendFromAlwaysSucceeds vhash validator txId
 
+    test "runPlutipContract: AlwaysSucceeds PlutusV2" do
+      let
+        distribution :: InitialUTxOs
+        distribution =
+          [ BigInt.fromInt 5_000_000
+          , BigInt.fromInt 2_000_000_000
+          ]
+      runPlutipContract config distribution \alice -> do
+        withKeyWallet alice do
+          validator <- AlwaysSucceedsV2.alwaysSucceedsScriptV2
+          let vhash = validatorHash validator
+          logInfo' "Attempt to lock value"
+          txId <- AlwaysSucceeds.payToAlwaysSucceeds vhash
+          awaitTxConfirmed txId
+          logInfo' "Try to spend locked values"
+          AlwaysSucceeds.spendFromAlwaysSucceeds vhash validator txId
+
+    test "runPlutipContract: AlwaysFails Ada Collateral Return" do
+      let
+        distribution :: InitialUTxOs /\ InitialUTxOs
+        distribution =
+          [ BigInt.fromInt 10_000_000
+          , BigInt.fromInt 2_000_000_000
+          ] /\ [ BigInt.fromInt 2_000_000_000 ]
+      runPlutipContract config distribution \(alice /\ seed) -> do
+        validator <- AlwaysFails.alwaysFailsScript
+        let vhash = validatorHash validator
+        txId <- withKeyWallet seed do
+          logInfo' "Attempt to lock value"
+          txId <- AlwaysFails.payToAlwaysFails vhash
+          awaitTxConfirmed txId
+          pure txId
+
+        withKeyWallet alice do
+          awaitTxConfirmed txId
+          logInfo' "Try to spend locked values"
+          balanceBefore <- fold <$> getWalletBalance
+          AlwaysFails.spendFromAlwaysFails vhash validator txId
+          balance <- fold <$> getWalletBalance
+          let
+            collateralLoss = Value.lovelaceValueOf $ BigInt.fromInt $ -5_000_000
+          balance `shouldEqual` (balanceBefore <> collateralLoss)
+
+    test "runPlutipContract: AlwaysFails Native Asset Collateral Return" do
+      let
+        distribution :: InitialUTxOs /\ InitialUTxOs
+        distribution =
+          [] /\ [ BigInt.fromInt 2_100_000_000 ]
```

Comment:
Originally used a separate wallet for creating the initial utxos here to keep alices numbers round, although I moved balance checking to be relative to the start so not really necessary any more